### PR TITLE
ci: buildkite release pipeline

### PR DESCRIPTION
**Pull request by v1v**
## Summary

Support Buildkite release pipeline

## Implementation details

Sign the artifacts using Buildkite

- [x] Support for Macos sign
- [x] Support for Windows sign
- [x] Support for *Nix sign

## How to validate this change

In Buildkite

## Test

> 🔈 Only for Elastic employees

[Build 26](https://buildkite.com/elastic/synthetics-recorder-release/builds/26) ran a dry-run (so it signed all the artifacts but didn't upload them to s3 or GitHub releases.

### Tasks

- [x] S3 bucket with some include files otherwise, it will upload even some nested folders.

### Follow-ups

- Remove Jenkins implementation

### CODE REVIEW

File: .buildkite/release.yml [hunk 1]
```diff
@@ -2,40 +2,66 @@
...
 
-  - label: "Sign windows :windows:"
-    command: "true"
+  - label: Windows Sign artifacts
+    trigger: unified-release-windows-signing
+    # NOTE: If you change 'key: windows-sign-service' then change the bellow call for .buildkite/scripts/download-signed-artifacts.sh
+    key: windows-sign-service
+    build:
+      env:
+        DOWNLOAD_ARTIFACTS_FILTER : "*.exe"
```
Comment (v1v) [inline .buildkite/release.yml:36]:
```suggestion
        DOWNLOAD_ARTIFACTS_FILTER: "*.exe"
```

File: .buildkite/release.yml [hunk 2]
```diff
@@ -2,40 +2,64 @@
...
+        INPUT_PATH: buildkite://
+
+  - wait: ~
+
+  - label: Download signed artifacts
+    commands:
+      - .buildkite/scripts/download-signed-artifacts.sh "gpg-sign-service"
+      - .buildkite/scripts/download-signed-artifacts.sh "macos-sign-service"
+      - .buildkite/scripts/download-signed-artifacts.sh "windows-sign-service"
+      - .buildkite/scripts/download-signed-artifacts.sh "gpg-sign-service | buildkite-agent pipeline upload
```
Comment (v1v) [inline .buildkite/release.yml:44]:
```suggestion
      - .buildkite/scripts/download-signed-artifacts.sh "gpg-sign-service" | buildkite-agent pipeline upload
```

File: .buildkite/release.yml [hunk 3]
```diff
@@ -2,40 +2,70 @@
...
+      - .buildkite/scripts/create-github-release.sh
+    depends_on:
+      - "gpg"
+      - "macos"
+      - "windows"
 
+# Figure out how to notify releases (maybe subscribe the slack channel to the GitHub releases?)
 notify:
-  - slack: "#observablt-bots"
+  - slack: "#on-week-oblt-productivity"
```
Comment (v1v) [inline .buildkite/release.yml:70]: What channel?
```suggestion
  - slack: "#on-week-oblt-productivity"
```

File: .buildkite/release.yml [hunk 4]
```diff
@@ -2,40 +2,70 @@
...
 
   - wait: ~
 
   - label: "Publish S3 Artifacts :s3:"
-    command: "true"
+    trigger: unified-release-publish-s3-artifacts
+    key: publish-s3-service
+    build:
+      env:
+        DESTINATION_PATH: "s3://download.elasticsearch.org/synthetics-recorder/"
```
Comment (v1v) [inline .buildkite/release.yml:55]: need to exclude the subfolders but use the files in the root directory. Waitign for the RElease team to provide this feature

File: .buildkite/hooks/pre-command [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+##  This script prepares the secret context
+##
+##  NOTE: *_SECRET or *_TOKEN env variables are masked, hence if you'd like to avoid any
+##        surprises please use the suffix _SECRET or _TOKEN for those values that contain
+##        any sensitive data. Buildkite can mask those values automatically
+
+set -eo pipefail
+
+# To help with testing the GPG signing
+BUILDKITE_TOKEN_SECRET=$(vault kv get -field=buildkite_token kv/ci-shared/observability-ci/buildkite-read-build-access)
```
Comment (v1v) [inline .buildkite/hooks/pre-command:11]: There is an ongoing task to move away from BK API Tokens and use some opinionated contracts so we can skip this API call for the sake of simplicity. Release team is working on it

**Review by reakaleek — APPROVED** [review]

File: .buildkite/scripts/create-github-release.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  gh release \
+    create \
+    "${BUILDKITE_TAG}" \
+    --draft \
+    --title "${BUILDKITE_TAG}" \
+    --repo elastic/synthetics-recorder \
+    "${DIST_LOCATION}/*.*"
+else
+  echo "gh release won't be triggered this is not a Git tag release"
+fi
```
Comment (reakaleek) [inline .buildkite/scripts/create-github-release.sh:38]: nit: wondering if it's just easier to create a release directly with

```
curl -L \
  -X POST \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <YOUR-TOKEN>" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/OWNER/REPO/releases \
  -d '{"tag_name":"v1.0.0","target_commitish":"master","name":"v1.0.0","body":"Description of the release","draft":false,"prerelease":false,"generate_release_notes":false}'
  ```

Comment (v1v) [reply .buildkite/scripts/create-github-release.sh:38]: I copied the script from https://github.com/elastic/synthetics-recorder/blob/main/.ci/scripts/create-github-release.sh.

In addition, it does upload the artifacts too, see `"${DIST_LOCATION}/*.*"`

Comment (reakaleek) [reply .buildkite/scripts/create-github-release.sh:38]: Ah, right. lgtm

**Review by justinkambic — APPROVED** [review]
LGTM

**Comment by v1v** [comment]
Unfortunately tests are broken :/

**Review by reakaleek — APPROVED** [review]

**Comment by v1v** [comment]
Thanks @justinkambic , I'm afraid the CI is broken, but no clue what's the error about :/ @elastic/uptime , can you help with this?

**Pull request merged**